code/network/multimsgs.cpp

```diff
@@ -2755,17 +2755,17 @@ void process_weapon_kill_packet(ubyte *data, header *hinfo )
 		Weapons[missile->instance].weapon_flags.set(Weapon::Weapon_Flags::Destroyed_by_weapon);
 	}
 	
-	Weapons[missile->instance].lifeleft = -1.0f;
-	if (Weapon_info[Weapons[missile->instance].weapon_info_index].weapon_hitpoints > 0.0f) {
-		missile->hull_strength = -1.0f;
-	}
 
-	// kill the weapon!
+	// kill the missile! -- If it not a missile, ignore it, since they shouldn't be here anyway.
 	if (Weapon_info[Weapons[missile->instance].weapon_info_index].subtype & WP_MISSILE) {
```

Comment:
If this packet is for missiles only then why is this check even needed? There should be a safety check in the `send_*()` function which prevents it from sending if not a missile, and perhaps one in the `process_*()` function to bail early too. The point of the packet shouldn't be under an if-statement though since that's sort of confusing.